notification-service/src/test/java/com/redhat/parodos/notification/service/NotificationRecordServiceTests.java

```diff
@@ -0,0 +1,232 @@
+
+		Mockito.reset(this.notificationRecordRepository);
+
+		// Notification User exists, state empty and searchTerm valid
+		Mockito.when(this.notificationUserRepository.findByUsername(userName)).thenReturn(validOptional);
+		Mockito.when(this.notificationRecordRepository.search(notificationUser, searchTerm.toLowerCase(), pageable))
+				.thenReturn(expectedResult);
+		result = this.notificationRecordServiceImpl.getNotificationRecords(pageable, userName, null, searchTerm);
+		assertEquals(expectedResult, result);
+		Mockito.verify(this.notificationRecordRepository, Mockito.times(1)).search(notificationUser,
+				searchTerm.toLowerCase(), pageable);
+
+		Mockito.reset(this.notificationRecordRepository);
+
+		// Notification User exists, state and searchTerm valid - exception expected
+		Mockito.when(this.notificationUserRepository.findByUsername(userName)).thenReturn(validOptional);
+		exception = assertThrows(RuntimeException.class, () -> {
+			this.notificationRecordServiceImpl.getNotificationRecords(pageable, userName, State.UNREAD, searchTerm);
+		});
+		assertEquals("Search by state and search term combined not supported", exception.getMessage());
+	}
+
+	@Test
+	void countNotificationRecords() {
+		String userName = "test-user";
+		List<State> states = Arrays.asList(State.values());
+
+		for (State state : states) {
+			if (State.UNREAD.equals(state)) {
+				// Test Notification User exists with 2 records
+				NotificationUser notificationUser = Mockito.mock(NotificationUser.class);
+				NotificationRecord notificationRecord = Mockito.mock(NotificationRecord.class);
+				Optional<NotificationUser> validOptional = Optional.of(notificationUser);
+				Mockito.when(this.notificationUserRepository.findByUsername(userName)).thenReturn(validOptional);
+				Mockito.when(this.notificationRecordRepository
+						.countDistinctByReadFalseAndNotificationUserListContaining(notificationUser)).thenReturn(2);
+				List<NotificationRecord> records = Arrays.asList(notificationRecord);
+				Mockito.when(notificationUser.getNotificationRecordList()).thenReturn(records);
+				assertEquals(2, this.notificationRecordServiceImpl.countNotificationRecords(userName, state));
+
+				// Test Notification User doesn't exist
+				Optional<NotificationUser> emptyOptional = Optional.empty();
+				Mockito.when(this.notificationUserRepository.findByUsername(userName)).thenReturn(emptyOptional);
+				assertEquals(0, this.notificationRecordServiceImpl.countNotificationRecords(userName, state));
+			}
+			else {
+				// Test state value is not "UNREAD", exception is expected
+				Exception exception = assertThrows(RuntimeException.class, () -> {
+					this.notificationRecordServiceImpl.countNotificationRecords(userName, state);
+				});
+				assertEquals(String.format("State %s is not supported", state), exception.getMessage());
+			}
+		}
+	}
+
+	@Test
+	void updateNotificationStatus() {
+		List<Operation> operations = Arrays.asList(Operation.values());
```

Comment:
`Arrays.asList` is unnecessary here

Reply:
Removed this line and used the `Operation.values()` in the foreach loop